utils/ordered_map.go

```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	"cmp"
+	"iter"
+	"slices"
+)
+
+func OrderMap[K cmp.Ordered, T any](m map[K]T) iter.Seq2[K, T] {
+	// 1. collect keys
+	keys := make([]K, 0, len(m))
```

Comment:
I've seen we use `maps.Keys` e.g. in state_updates, wonder if we could use it here as well?

Reply:
Rewrote it with `maps.Keys(...)` and `slices.Sorted(...)`. Former action actually return an iterator, but that's ok